mlir/python/mlir/dialects/_ods_common.py

```diff
@@ -123,3 +130,52 @@ def get_op_result_or_op_results(
         if len(op.results) > 0
         else op
     )
+
+
+U = _TypeVar("U", bound=_cext.ir.Value)
```

Comment:
Should this be made module-private, `_U`? I'm not sure we want to have this exported.